docs/misc/config.md

```diff
@@ -0,0 +1,95 @@
+
+#### List Values
+
+Config values that can take in a list of entries can be defined using `#defineList`. If the list provided by the consumer can be empty, `#defineListAllowEmpty` should be used instead. The method is broken down into three components: a path representing the name of the variable, the default value when no valid configuration is present, and a validator to make sure a deserialized element from the list is valid.
+
+#### Enum Values
+
+Config values that represent an `Enum` can be defined by `EnumValue`s via `#defineEnum`. The method is broken down into four components: a path representing the name of the variable, the default value when no valid configuration is present, a getter to convert a string or integer into an enum, and a collection of the allowed values the configuration can be.
+
+#### Boolean Values
+
+Config values that represent a `boolean` can be defined by `BooleanValue`s via `#define`. The method is broken down into two components: a path representing the name of the variable and the default value when no valid configuration is present.
+
+#### Double Values
+
+Config values that represent a `double` can be defined by `DoubleValue`s via `#defineInRange`. The method is broken down into four components: a path representing the name of the variable, the default value when no valid configuration is present, and the minimum and maximum value.
+
+#### Int Values
+
+Config values that represent an `int` can be defined by `IntValue`s via `#defineInRange`. The method is broken down into four components: a path representing the name of the variable, the default value when no valid configuration is present, and the minimum and maximum value.
+
+#### Long Values
+
+Config values that represent a `long` can be defined by `LongValue`s via `#defineInRange`. The method is broken down into four components: a path representing the name of the variable, the default value when no valid configuration is present, and the minimum and maximum value.
+
+Registering a Configuration
+---------------------------
+
+Once a `ForgeConfigSpec` has been built, it must be registered to allow Forge to load, track, and sync the configuration settings as required. Configurations should be registered in the mod constructor via `ModLoadingContext#registerConfig`. A configuration can be registered with a given type representing the side the config belongs to (`CLIENT`, `COMMON`, `SERVER`), the `ForgeConfigSpec`, and optionally a specific file name for the configuration.
+
+```java
+// In the mod constructor with a ForgeConfigSpec CONFIG
+ModLoadingContext.get().registerConfig(Type.COMMON, CONFIG);
+```
+
+!!! tip
+    The definition of each config type and what they are used for can be found as part of the [constant's javadoc][type].
```

Comment:
As the details of each config type are important to the selection and behavior of the config, those types should be documented here rather than linking to the javadocs.